src/Elliptic/BoundaryConditions/AnalyticSolution.hpp

```diff
@@ -0,0 +1,234 @@
+ private:
+  using Base = BoundaryCondition<Dim, Registrars>;
+
+ public:
+  using options =
+      tmpl::list<elliptic::OptionTags::BoundaryConditionType<FieldTags>...>;
+  static constexpr Options::String help =
+      "Boundary conditions from the analytic solution";
+
+  AnalyticSolution() = default;
+  AnalyticSolution(const AnalyticSolution&) noexcept = default;
+  AnalyticSolution& operator=(const AnalyticSolution&) noexcept = default;
+  AnalyticSolution(AnalyticSolution&&) noexcept = default;
+  AnalyticSolution& operator=(AnalyticSolution&&) noexcept = default;
+  ~AnalyticSolution() noexcept = default;
+
+  /// \cond
+  explicit AnalyticSolution(CkMigrateMessage* m) noexcept : Base(m) {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(AnalyticSolution);
+  /// \endcond
+
+  /// Select which `elliptic::BoundaryConditionType` to apply for each field
+  explicit AnalyticSolution(
+      const typename elliptic::OptionTags::BoundaryConditionType<
+          FieldTags>::type... boundary_condition_types) noexcept
+      : boundary_condition_types_{boundary_condition_types...} {}
+
+  std::unique_ptr<domain::BoundaryConditions::BoundaryCondition> get_clone()
+      const noexcept override {
+    return std::make_unique<AnalyticSolution>(*this);
+  }
+
+  const auto& boundary_condition_types() const noexcept {
+    return boundary_condition_types_;
+  }
+
+  using argument_tags =
+      tmpl::list<::Tags::AnalyticSolutionsBase, domain::Tags::Mesh<Dim>,
+                 domain::Tags::Direction<Dim>,
+                 ::Tags::Normalized<domain::Tags::UnnormalizedFaceNormal<
+                     Dim, Frame::Inertial>>>;
+  using volume_tags =
+      tmpl::list<::Tags::AnalyticSolutionsBase, domain::Tags::Mesh<Dim>>;
+
+  template <typename OptionalAnalyticSolutions>
+  void apply(const OptionalAnalyticSolutions& optional_analytic_solutions,
+             const Mesh<Dim>& volume_mesh, const Direction<Dim>& direction,
+             const tnsr::i<DataVector, Dim>& face_normal,
+             const gsl::not_null<typename FieldTags::type*>... fields,
+             const gsl::not_null<typename FieldTags::type*>... n_dot_fluxes)
+      const noexcept {
+    const auto& analytic_solutions =
+        [&optional_analytic_solutions]() noexcept -> decltype(auto) {
+      if constexpr (tt::is_a_v<std::optional, OptionalAnalyticSolutions>) {
+        if (not optional_analytic_solutions.has_value()) {
+          ERROR(
+              "The background does not provide an analytic solution. Select an "
+              "analytic solution as background if you want to use it to impose "
+              "boundary conditions.");
+        }
+        return *optional_analytic_solutions;
+      } else {
+        return optional_analytic_solutions;
+      }
+    }();
+    const size_t slice_index =
+        index_to_slice_at(volume_mesh.extents(), direction);
+    const auto helper = [this, &analytic_solutions, &volume_mesh, &direction,
```

Comment:
`helper` -> `...` whatever this actually does. Maybe `data_on_boundary` or something like that